### PR TITLE
Partially revert PR #6384 for URLs using a git commit hash, and make the `native_cctools` tarballs download to meaningful names

### LINKED ISSUE #6415: New download paths for github-based depends don't seem to work when the package version is a git hash
For `native_cctools`, we have
```
package=native_cctools
$(package)_version=55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
$(package)_download_path=https://github.com/tpoechtrager/cctools-port/archive/refs/tags
$(package)_file_name=$($(package)_version).tar.gz
$(package)_sha256_hash=e51995a843533a3dac155dd0c71362dd471597a2d23f13dff194c6285362f875
```
in `packages/native_cctools.mk`, but if we try and fetch what I believe to be the URL, it fails with a 404:
```
sasha_work@seasonal-dream compares % wget https://github.com/tpoechtrager/cctools-port/archive/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
--2023-02-08 13:48:29--  https://github.com/tpoechtrager/cctools-port/archive/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97 [following]
--2023-02-08 13:48:29--  https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-02-08 13:48:29 ERROR 404: Not Found.

sasha_work@seasonal-dream compares % 
```

Likewise with `tl_expected`:

```
package=tl_expected
$(package)_version=1.0.1
$(package)_download_path=https://github.com/TartanLlama/expected/archive/refs/tags
$(package)_download_file=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
$(package)_file_name=tl-expected-1.0.1.tar.gz
$(package)_sha256_hash=64901df1de9a5a3737b331d3e1de146fa6ffb997017368b322c08f45c51b90a7
```

gets us:
```
sasha_work@seasonal-dream compares % wget https://github.com/TartanLlama/expected/archive/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
--2023-02-08 13:49:13--  https://github.com/TartanLlama/expected/archive/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/TartanLlama/expected/tar.gz/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5 [following]
--2023-02-08 13:49:13--  https://codeload.github.com/TartanLlama/expected/tar.gz/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-02-08 13:49:14 ERROR 404: Not Found.

sasha_work@seasonal-dream compares % 
```

The old path (with the `refs/tags` excised) works for both:

```
sasha_work@seasonal-dream compares % wget https://github.com/TartanLlama/expected/archive/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz 
--2023-02-08 13:49:51--  https://github.com/TartanLlama/expected/archive/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/TartanLlama/expected/tar.gz/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5 [following]
--2023-02-08 13:49:51--  https://codeload.github.com/TartanLlama/expected/tar.gz/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz.2’

96d547c03d2feab8db64c53c3744a9b4a7c8f2     [ <=>                                                                        ]  21.48K  --.-KB/s    in 0.001s  

2023-02-08 13:49:51 (33.1 MB/s) - ‘96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz.2’ saved [21994]

sasha_work@seasonal-dream compares % wget https://github.com/tpoechtrager/cctools-port/archive/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz 
--2023-02-08 13:50:01--  https://github.com/tpoechtrager/cctools-port/archive/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97 [following]
--2023-02-08 13:50:01--  https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.2’

55562e4073dea0fbfd0b20e0bf69ffe6390c7f     [   <=>                                                                      ]   2.72M  5.82MB/s    in 0.5s    

2023-02-08 13:50:02 (5.82 MB/s) - ‘55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.2’ saved [2851747]

sasha_work@seasonal-dream compares % 
```

I believe this change was introduced with PR #6384; and there might be something I am missing (maybe the URL derivation process is not what I think it is).

### PR DESCRIPTION
**Pull request by softminus**
The new URLs (with `archive/ref/tags`) are good for tag-based references so we leave them alone, and only revert the hash-based URLs to GitHub URLs which doesn't 404 and give us files with the expected hashes today.

Currently, the three repos in question (`cctools-port`, `apple-libtapi`, and `expected`) do not have usable tags/releases we can use instead of the hashes.

We will also upload correct versions of those files to the download.z.cash server (with the correct filenames) so if there's another issue with hash stability (like there was on 30 January 2023 https://github.com/orgs/community/discussions/46034) it won't instantly break the build.

This should fix #6415.

@str4d suggested changing `native_cctools.mk` so that the saved filenames for `native_cctools[_libtapi]` aren't just long hash names. @daira also pointed out that we had used a non-existent version number in the saved filename for `tl_expected`. This PR fixes these issues, saving to `native_cctools_55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz`, `native_cctools_libtapi_3efb201881e7a76a21e0554906cf306432539cef.tar.gz`, and `tl_expected_96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz`. These download OK with both the github URLs and also (by breaking the github download URL to force it) download OK with the download.z.cash server.

Since the filename on the download.z.cash server needs to match the *saved* filename (and not the filename in the original URL, which happened to coincide with the saved filename for the two tarballs in `native_cctools.mk`), we needed to reupload copies to the download server with the right naming scheme.

### CODE REVIEW

**Review by daira — DISMISSED** [review]
utACK

**Comment by softminus** [comment]
@str4d suggested changing `native_cctools.mk` so the saved filenames aren't just a long hash name, I marked this PR draft so we can do that as well. Since the filename on the download.z.cash server needs to match the *saved* filename (and not the filename in the original URL, which happened to coincide with the saved filename for the two tarballs in `native_cctools.mk`), we'll need to reupload copies to the download server with the right naming scheme.

**Comment by softminus** [comment]
Also the two `native_cctools` tarballs seem to download and extract OK on my linux machine (I disabled the clang depend for quick testing, if I enable it it still doesn't build OK on my linux box and fails with a different error) -- both when the github tarball is fetched and when the tarball is fetched from download.z.cash:

```
CMake Error at /usr/share/cmake-3.22/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  /home/sasha_work/src/zcash/depends/x86_64-pc-linux-gnu/native/bin/clang.
Call Stack (most recent call first):
  CMakeLists.txt:45 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_ASM_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/home/sasha_work/src/zcash/depends/work/build/x86_64-pc-linux-gnu/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-110c8bdc57d/libtapi/build/CMakeFiles/CMakeOutput.log".
make: *** [funcs.mk:278: /home/sasha_work/src/zcash/depends/work/build/x86_64-pc-linux-gnu/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-110c8bdc57d/cctools/.stamp_configured] Error 1
sasha_work@myouren:~/src/zcash/depends$ ls work/
build/    download/ 
sasha_work@myouren:~/src/zcash/depends$ ls work/build/x86_64-pc-linux-gnu/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-110c8bdc57d/libtapi/
build  build.sh  install.sh  LICENSE.APPLE-LIBTAPI.txt  LICENSE.LLVM.txt  README.md  src  tools  VERSION.txt
sasha_work@myouren:~/src/zcash/depends$ ls work/build/x86_64-pc-linux-gnu/native_cctools/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97-110c8bdc57d/
cctools  libtapi  package.sh  README.md  tools  usage_examples
sasha_work@myouren:~/src/zcash/depends$ 
```

File: depends/packages/tl_expected.mk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package=tl_expected
 $(package)_version=1.0.1
```
Comment (daira) [inline depends/packages/tl_expected.mk:2]:
```suggestion
$(package)_version=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
```

File: depends/packages/tl_expected.mk [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-$(package)_download_path=https://github.com/TartanLlama/expected/archive/refs/tags
+$(package)_download_path=https://github.com/TartanLlama/expected/archive
 $(package)_download_file=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
 $(package)_file_name=tl-expected-1.0.1.tar.gz
```
Comment (daira) [inline depends/packages/tl_expected.mk:5]:
```suggestion
$(package)_download_file=$($(package)_version).tar.gz
$(package)_file_name=$(package)_$($(package)_version).tar.gz
```
No 1.0.1 version has been released, and it would be at a different commit if it were.

**Review by daira — APPROVED** [review]
utACK. I changed `tl_expected` to use the same filename conventions as `native_cctools` (third commit).

**Comment by softminus** [comment]
looks good to me, @daira, I'm gonna upload the new filename for `tl_expected` so `https://download.z.cash/depends-sources/tl_expected_96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz` doesn't 404 and I can test it both ways

**Comment by softminus** [comment]
Seems to download OK both via github and the download server!

**Comment by daira** [comment]
Taking @softminus' testing as a review of the third commit, this has sufficient review.
